### DISCUSSION TITLE
Error message while running Mfront on Ubuntu

### DISCUSSION THREAD

**Discussion by Nadia-Dempowo** [thelfer/tfel#609] [General]
Hi,
I tried to run an Mfront file on Ubuntu but I got this message. 

./mfront: error while loading shared libraries: libTFELMFront.so.4.2.0: cannot open shared object file: No such file or directory

Could you please explain the cause of this error and how to resolve it?

Best regards,
Nadia

**Comment by thelfer** [comment]
Hi @Nadia-Dempowo,
The error message probably means that you did not update your `LD_LIBRARY_PATH` variable. Here is the environment file that I use:

~~~~bash
export TFELHOME=/home/th202608/codes/tfel/4.2.0/install-python-3.10.12
export PATH=$TFELHOME/bin:$PATH
export LD_LIBRARY_PATH=$TFELHOME/lib:$LD_LIBRARY_PATH
export PYTHONPATH=$TFELHOME/lib/python3.10/site-packages/:$PYTHONPATH
~~~~

Of course, you need to adapt the paths and you may not need to update the `PYTHONPATH` if you did not build `TFEL` with the `python` bindings. 

I write those lines in a file named `env-4.2.0.sh` and type:

```bash
source env-4.2.0.sh
```

everytime I need to use `MFront 4.2`

**Reply by Nadia-Dempowo** [reply]
Hi @thelfer,

Thanks for your feedback. But when I add the "export" lines above to a file and do "source" as you described, I still get the same error. I've also indicated the LIBRAIRY_PATH in the .bashrc file I've attached, but the error persists. 
I want to understand something please: in your answer, you talk about building TFEL with python. From what I read in the Mfront documentation, there ' s no need for additional installation of Mfront for all versions of Mfront after 2019. So when I installed Castem2024, I thought Mfront is automatically installed too. Or maybe I've misunderstood? Please explain if there are any additional installations to be done? If so, which ones? 
Please find attached a screenshot of the LIBRARY_PATH update.
![LD_LIBRARY_PATH](https://github.com/user-attachments/assets/6dce35e9-150f-433d-8007-de66acde8412)

Thank you,

Nadia

**Reply by Nadia-Dempowo** [reply]
Hi @thelfer, 

I hadn't properly modified the export paths. I changed the paths and now I'm running my Mfront file. Thanks for your answer. 

Thanks a lot,
Nadia

**Reply by thelfer** [reply]
Great. Let me know if you need more help !

**Comment by Nadia-Dempowo** [comment]
Hi @thelfer ,
I wrote a few weeks ago about installing Mfront; everything works. thanks again.
I have another question today about using Mfront. Sorry, I'm a new Mfront user, so the question is probably a basic one. In fact, I wrote an Mfront file to take into account variable properties based on temperatures and many others..., and I used it in Castem. The Mfront file works fine on its own and provides me with the libCastemUO2.so library, but when I use it in Castem, it seems that Castem doesn't recognize it and prints an error. Attached is an image of the mfront file and the castem file along with the error message. I don't really know if this is a bad implementation in castem as I implemented it using a table as shown in the documentation I saw on the Mfront site. Maybe I made a mistake somewhere, but I've been checking for a long time where I made errors and I can't find any, but it doesn't work. thanks 
![Erreur_affichee_par_castem](https://github.com/user-attachments/assets/a694a5cb-4541-4fec-b064-7250d1705f66)
![MfrontInCastemfile](https://github.com/user-attachments/assets/e5194609-265e-4782-852d-b7512de71cf1)

Thanks,
Nadia

**Reply by thelfer** [reply]
What version of Cast3M are you using ?
Thomas

**Reply by Nadia-Dempowo** [reply]
I am using Castem 2024

**Reply by Nadia-Dempowo** [reply]
I use this command in mfront:
mfront --obuild --interface=castem ThermalConductivityUO2.mfront

**Reply by thelfer** [reply]
I think that I got the issue. Nonlinear material properties are not meant to be passed to `COND` directly. In a nonlinear resolution, using `PASAPAS` for instance, the properties are first evaluated using the `VARI` operator and the resulting field is passed to `COND` (used for the linearisation of the nonlinear problem).

**Reply by Nadia-Dempowo** [reply]
Ah, okay,thank you for your prompt reply. I'll change it right now.

**Reply by Nadia-Dempowo** [reply]
Hello @thelfer , 
Just to let you know that I tested it and it works. As you said, you don't have to pass it directly to the COND operator. it works fine, thanks again for your help.
